File: python/7 kyu/57.py
# https://www.codewars.com/kata/57ae18c6e298a7a6d5000c7a/train/python

# Passed

def replace_all(obj, find, replace):
    items = list(obj) if type(obj) == str else obj
    for idx, item in enumerate(items):
        if item == find:
            items[idx] = replace
            
    if type(obj) == str:
        res = ''.join(items)
    else:
        res = obj
    
    return res

output = replace_all([1, 1, 2], 1, 2)
print(output)        

File: nodejs/7 kyu/57.js
// https://www.codewars.com/kata/5b16490986b6d336c900007d/train/javascript

// Passed

function myLanguages(results) {
  const reverseMap = {};
  Object.keys(results).forEach((lang) => {
    let score = results[lang];
    reverseMap[score] = lang;
  });

  const scores = Object.values(results)
    .sort((a, b) => b - a)
    .filter((score) => score >= 60);
  const languages = scores.map((score) => reverseMap[score]);
  return languages;
}

const output = myLanguages({ Java: 10, Ruby: 80, Python: 65 });
console.log(output);
